mesonbuild/build.py

```diff
@@ -3076,3 +3087,97 @@ def save(obj: Build, filename: str) -> None:
             pickle.dump(obj, f)
     finally:
         obj.environment.coredata = cdata
+
+
+@dataclass(eq=False)
+class ResponseFile(HoldableObject):
+
+    args: T.List[CommandArg]
+    subdir: str
+    environment: environment.Environment
+
+    prefix: str = '@'
+    quote: T.Union[bool, str] = True
+    max_args_length: T.Optional[int] = None
+    separator: str = ' '
+    relative_paths: bool = False
+
+    def __post_init__(self):
+        self._response_file: T.Optional[Path] = None
+
+        if self.max_args_length is None:
+            self.max_args_length = get_rsp_threshold()
+
+        self.args = [self.processStringArg(a) for a in self.args]
+
+    def has_response_file(self) -> bool:
+        return self._response_file is not None
+
+    def configure(self, filename: str) -> None:
+        if not self.args:
+            return
+
+        quoted_args = self.separator.join(self.quote_fct(a) for a in self.generateArgList(self.args))
+        if len(quoted_args) <= self.max_args_length:
+            mlog.debug(f'Estimated arg length {len(quoted_args)} <= {self.max_args_length}')
+            return
+
+        self._response_file = Path(self.subdir, filename).with_suffix('.rsp')
+
+        mlog.debug(f'Writing rsp file: {self._response_file}')
+
+        response_file_fullpath = Path(self.environment.build_dir, self._response_file)
+        response_file_fullpath.parent.mkdir(exist_ok=True)
+        response_file_fullpath.write_text(quoted_args, encoding='utf-8')
+
+    def generateArgList(self, args: T.List[CommandArg]) -> Generator[FileOrString, None, None]:
```

Comment:
Don't use camel casing for function

Reply:
This also needs a return type.

Reply:
> This also needs a return type.

isn't the `Generator` already a return type?